test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -52,6 +54,27 @@ TEST(validator_test, fullfill_concept)
     EXPECT_TRUE(validator_concept<file_ext_validator>);
 }
 
+TEST(validator_test, no_file)
+{
+    filesystem::path p{"./sandbox.fasta"};
+    std::string s{"./stonebox.fasta"};
+    file_existance_validator my_validator{};
+    EXPECT_THROW(my_validator(p), parser_invalid_argument);
+    EXPECT_THROW(my_validator(s), parser_invalid_argument);
+}
+
+TEST(validator_test, file_exists)
+{
+    test::tmp_filename("./sandbox2.fasta");
+    filesystem::path p{"./"};
+    std::string s{"./"};
+    file_existance_validator my_validator{};
+    for (auto & file : filesystem::directory_iterator(p))
```

Comment:
You need to actually use the `tmp_filename` to create a file that exists! 
```cpp
test::tmp_filename tmp_file_name{"./sandbox2.fasta"};
std::ofstream tmp_file(tmp_file_name.get_path());
tmp_file << "foo\n"; // I am actually not sure if you need this line because the constructor may already create the file
```
and then 
```suggestion
    for (auto & file : filesystem::directory_iterator(tmp_filename.get_path()))
```

Reply:
@MitraDarja ping :)